### PR TITLE
Add playback rate command line arg

### PR DESCRIPTION
**Pull request by mabelzhang**
Addresses #290

Both `-r` and `-s` were taken, so I wasn't sure what is the next best letter to use for rate / speed. I just put in `-x` for this first implementation.

Examples of usage:
```
$ ros2 bag play -x 2.5 rosbag2_2020_01_15-21_25_18
$ ros2 bag play -x 0.5 rosbag2_2020_01_15-21_25_18
$ ros2 bag play -x 1 rosbag2_2020_01_15-21_25_18
```

with updated usage message:
```
$ ros2 bag play -h
usage: ros2 bag play [-h] [-s STORAGE] [-r READ_AHEAD_QUEUE_SIZE] [-x RATE]
                     bag_file

ros2 bag play

positional arguments:
  bag_file              bag file to replay

optional arguments:
  -h, --help            show this help message and exit
  -s STORAGE, --storage STORAGE
                        storage identifier to be used, defaults to "sqlite3"
  -r READ_AHEAD_QUEUE_SIZE, --read-ahead-queue-size READ_AHEAD_QUEUE_SIZE
                        size of message queue rosbag tries to hold in memory
                        to help deterministic playback. Larger size will
                        result in larger memory needs but might prevent delay
                        of message playback.
  -x RATE, --rate RATE  rate at which to play back messages
```

Signed-off-by: Mabel Zhang <mabel@openrobotics.org>

### CODE REVIEW

**Comment by eric1221bday** [comment]
So this is speculation on my part, but what would happen if the user enters a negative rate, nothing in the options forbid this and it would I think result in the player playing all messages as fast as possible with the current implementation. Maybe add a check on the sign of the rate?

File: ros2bag/ros2bag/verb/play.py [hunk 1]
```diff
@@ -32,6 +32,9 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             help='size of message queue rosbag tries to hold in memory to help deterministic '
                  'playback. Larger size will result in larger memory needs but might prevent '
                  'delay of message playback.')
+        parser.add_argument(
+            '-x', '--rate', type=float, default=1.0,
```
Comment (emersonknapp) [inline ros2bag/ros2bag/verb/play.py]: not a big fan of `x` here, maybe we just want to only provide the long `--rate` option?

**Comment by Karsten1987** [comment]
just throwing in an idea here: We could also remove the `-r` from the read ahead queue. This would clearly break stuff, but I am also convinced that `-r` will be used more frequently for rate than for the read ahead queue.

**Comment by emersonknapp** [comment]
> -r will be used more frequently for rate than for the read ahead queue.

Agreed, i think it should be fine to have the queue be long-form only

File: ros2bag/ros2bag/verb/play.py [hunk 2]
```diff
@@ -32,6 +32,9 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
...
+            help='rate at which to play back messages')
```
Comment (Karsten1987) [inline ros2bag/ros2bag/verb/play.py]: consider making this more explicit that any value <=0.0 results in no messages being replayed.

**Comment by mabelzhang** [comment]
bbb4215 should address all the comments above -

- removed `-r` from `--read-ahead-queue-size` and use for `--rate`.
- added range > 0.0 check
- added tests

**Comment by Karsten1987** [comment]
@mabelzhang Can you run CI on this? I want to make sure we don't have regression on OSX and Windows with this.

**Comment by mabelzhang** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9545/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5251/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7797/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9469/)

**Comment by Karsten1987** [comment]
new round of CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9585/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5274/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7820/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9501/)

**Comment by mabelzhang** [comment]
Oh oops didn't see that. I just started another round with a better .repos file heh
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9586/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5275/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7821/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9502/)

**Comment by Karsten1987** [comment]
@mabelzhang you might have to rebase your work on top of master.

**Comment by mabelzhang** [comment]
New set of CIs. The previous one was building on eloquent branch for many repos, I must have copied from the wrong ros2.repos file.

Are the unstables expected?
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9609/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5304/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7832/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9514/)

**Comment by mabelzhang** [comment]
Finally
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9611/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5307/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7834/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9516/)

The Windows cmake just has this mysterious complaint that some other builds also have:
```
No project() command is present. The top-level CMakeLists.txt file must
```

**Comment by Karsten1987** [comment]
Can you point to that other project which also has this warning? Are you aware of any ticket tracking this issue already?

I just want to avoid any regression introduced by this patch.

**Comment by mabelzhang** [comment]
So the warning is this one:
https://ci.ros2.org/job/ci_windows/9516/cmake/new/
and the history graph looks like up and down from 0 to 1 alternately.

This earlier build with the same warning is just building plain ros2.repos with no additional flags:
https://ci.ros2.org/job/ci_windows/9512/cmake/new/

This one has the same warning too:
https://ci.ros2.org/job/ci_windows/9509/cmake/new/
This is the earliest one I see. There are other builds that say 1 warning in the graph, but when I click into them, they say 0, like 9506, 9500, 9499. I'm not sure why it's like that.

I don't know if a ticket is tracking this.

**Comment by Karsten1987** [comment]
@mabelzhang with #323 being merged, do you mind rebasing your branch and re-run CI? I am confident that CI should return all green then and this PR finally being ready to get merged ;)

**Comment by mabelzhang** [comment]
Hurray
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/9694/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5358/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/7908/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/9596/)

**Pull request merged**